pkg/quarterdeck/accounts.go

```diff
@@ -0,0 +1,101 @@
+package quarterdeck
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/oklog/ulid/v2"
+	"github.com/rotationalio/ensign/pkg/quarterdeck/api/v1"
+	"github.com/rotationalio/ensign/pkg/quarterdeck/db/models"
+	"github.com/rotationalio/ensign/pkg/quarterdeck/middleware"
+	"github.com/rotationalio/ensign/pkg/quarterdeck/tokens"
+	ulids "github.com/rotationalio/ensign/pkg/utils/ulid"
+)
+
+func (s *Server) AccountUpdate(c *gin.Context) {
+	//TODO: add functionality to update email
+	var (
+		err    error
+		userID ulid.ULID
+		user   *api.User
+		model  *models.User
+		claims *tokens.Claims
+	)
+
+	// Retrieve ID component from the URL and parse it.
+	if userID, err = ulid.Parse(c.Param("id")); err != nil {
+		c.Error(err)
+		c.JSON(http.StatusNotFound, api.ErrorResponse("user id not found"))
+		return
+	}
+
+	if err = c.BindJSON((&user)); err != nil {
+		c.Error(err)
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not parse request"))
+		return
+	}
+
+	// Sanity check: the URL endpoint and the user ID on the model match.
+	if !ulids.IsZero(user.UserID) && user.UserID.Compare(userID) != 0 {
+		c.Error(api.ErrModelIDMismatch)
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(api.ErrModelIDMismatch))
+		return
+	}
+
+	// Validate the request from the API side.
+	if err = user.ValidateUpdate(); err != nil {
+		c.Error(err)
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(err))
+		return
+	}
+
+	// Fetch the user claims from the request
+	if claims, err = middleware.GetClaims(c); err != nil {
+		c.Error(err)
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("user claims unavailable"))
+		return
+	}
+
+	//retrieve the orgID and userID from the claims and check if they are valid
+	orgID := claims.ParseOrgID()
+	requesterID := claims.ParseUserID()
+	if ulids.IsZero(orgID) || ulids.IsZero(requesterID) {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("invalid user claims"))
+		return
+	}
+
+	//check that the requesterID in the claims matches the userID on the model
```

Comment:
This check is pretty important because it prevents users from editing accounts of other users, can we add a comment to that effect here?